filters/include/pcl/filters/impl/random_sample.hpp

```diff
@@ -92,7 +92,7 @@ template<typename PointT>
 void
 pcl::RandomSample<PointT>::applyFilter (std::vector<int> &indices)
 {
-  unsigned N = static_cast<unsigned> (indices_->size ());
+  unsigned int N = static_cast<unsigned> (indices_->size ());
```

Comment:
One additional comment. Everything which is gonna be used as an index needs to be of `size_t` type. Please apply that change all over this method.

Reply:
Good point, this makes also some static_cast obsolete.